csrc/preseg_passes/consecutive_cast.cpp

```diff
@@ -20,6 +24,99 @@ bool isCast(Expr* expr) {
   return false;
 }
 
+// don't move meta operation when the its output is a fusion output, or has
+// multiple uses. In which case, we will have to duplicate the meta operation,
+// which might not be optimal. See PresegTest.FusionTestCastOptimizationMetaOp2
+bool isMovableMeta(Expr* expr) {
+  // TODO BroadcastOp is a moveable meta operation. We should enable it after
+  // matmul scheduler is updated to support the new pattern with matmul. See
+  // issue: https://github.com/NVIDIA/Fuser/issues/3665.
+  return (expr != nullptr && !expr->output(0)->isFusionOutput() &&
+          expr->output(0)->uses().size() == 1) &&
+      (expr->isOneOf<SqueezeOp, ViewOp>() || ir_utils::isSimpleTVSet(expr));
+}
+
+// replays meta operation on `new_in`. return the new output from replayed meta
+// operation
+Val* replayMetaOnNewInput(
```

Comment:
This is an added function to replay the meta operation on the new input.

Squeeze/Broadcast/Set are all simple, but replaying of `ViewOp` requires replaying the transform, which justify having a separate function.